File: interview questions/removeKthNodeFromEnd.js
class LinkedList {
  constructor(value) {
    this.value = value;
    this.next = null;
  }
}

function removeKthNodeFromEnd(head, k) {
  let first = head;
  let second = head;

  // traverse second pointer ahead by k nodes
  for (let i = 0; i < k; i++) second = second.next;

  // case of removing first node
  if (second === null) {
    head.value = head.next.value;
    head.next = head.next.next;
    return;
  }

  // traverse first and second pointers until second pointer hits last node
  while (second.next !== null) {
    first = first.next;
    second = second.next;
  }

  // remove node
  first.next = first.next.next;
}

File: interview questions/removeKthNodeFromEnd.py
class LinkedList:
    def __init__(self, value):
        self.value = value
        self.next = None


def removeKthNodeFromEnd(head, k):
    first = head


second = head

for i in range(k):
	    second = second.next

	if second == None:
		head.value = head.next.value
		head.next = head.next.next
		return

	while second.next is not None:
		first = first.next
		second = second.next

	first.next = first.next.next
	pass
